components/menu/MenuItem.tsx

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ForwardedRef} from 'react';
+import styled, {css} from 'styled-components';
+import classNames from 'classnames';
+
+import {MenuItemProps} from './Menu.types';
+
+const MenuItemLabel = styled.span`
+    font-size: 14px;
+    line-height: 20px;
+    margin: 0 8px 0 0;
+`;
+
+const MenuItemDescription = styled.span`
+    font-weight: 400;
+    font-size: 12px;
+    line-height: 16px;
+    color: rgba(var(--center-channel-color-rgb), 0.56);
+`;
+
+const LabelWrapper = styled.div<{leadingElement: boolean}>(
+    ({leadingElement}) => css`
+        display: flex;
+        flex-direction: column;
+        padding-left: ${leadingElement ? '10px' : 0};
```

Comment:
this feels overly complicated. This could be made simpler by using `flex` and the `gap` property on the parent component or by using `grid`

Reply:
I am not sure how its overly complicated, please elaborate?

Reply:
of course

If you are using a `display: flex` on the parent you can set the a `gap` (shorthand for `column-gap` and `row-gap`) as well which then renders a whitespace between two elements (columns or rows).

So when you are defining the property on the parent component each item will be separated by that gap.

You could even go one step further to define `display: grid` and a layout on the parent, which essentially reduces the manual addition of styling and wrapping elements to make that layout possible. This helps in reducing the clutter on the DOM as well, since we don't need the elements (and additional styling on them) to align the contents of it inside that layout.

Here are some useful links:
- [gap explanation](https://css-tricks.com/almanac/properties/g/gap/)
- [grid cheatsheet](https://css-tricks.com/snippets/css/complete-guide-grid/)

If you want (and have time for it) we can go over this in more detail